### PR TITLE
HLE/Applets: Implement ErrEula applet

### PR DESCRIPTION
**Pull request by mailwl**
Fix **Funky Barn 3D - hangs at intro** #1606

### CODE REVIEW

File: src/core/hle/applets/erreula.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+#include "core/hle/applets/applet.h"
+#include "core/hle/kernel/shared_memory.h"
+
+namespace HLE {
+namespace Applets {
+
+class ErrEula final : public Applet {
+public:
+    ErrEula(Service::APT::AppletId id): Applet(id), started(false) { }
```
Comment (lioncash) [inline src/core/hle/applets/erreula.h]:
```cpp
explicit ErrEula(Service::APT::AppletId id): Applet(id) { }
```

File: src/core/hle/applets/erreula.h [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    ResultCode StartImpl(const Service::APT::AppletStartupParameter& parameter) override;
+    void Update() override;
+    bool IsRunning() const override { return started; }
+
+    /// TODO(Subv): Find out what this is actually used for.
+    /// It is believed that the application stores the current screen image here.
+    Kernel::SharedPtr<Kernel::SharedMemory> framebuffer_memory;
+
+    /// Whether this applet is currently running instead of the host application or not.
+    bool started;
```
Comment (lioncash) [inline src/core/hle/applets/erreula.h]:
```cpp
bool started = false;
```

File: src/core/hle/applets/erreula.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+};
+
+}
```
Comment (lioncash) [inline src/core/hle/applets/erreula.h]: } // namespace Applets
} // namespace HLE

File: src/core/hle/applets/erreula.cpp [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    Service::APT::SendParameter(message);
+
+    started = false;
+    return RESULT_SUCCESS;
+}
+
+void ErrEula::Update() {
+}
+
+}
```
Comment (lioncash) [inline src/core/hle/applets/erreula.cpp]:
```cpp
} // namespace Applets
} // namespace HLE
```

File: src/core/hle/applets/erreula.h [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+class ErrEula final : public Applet {
+public:
+    explicit ErrEula(Service::APT::AppletId id): Applet(id) { }
+
+    ResultCode ReceiveParameter(const Service::APT::MessageParameter& parameter) override;
+    ResultCode StartImpl(const Service::APT::AppletStartupParameter& parameter) override;
+    void Update() override;
+    bool IsRunning() const override { return started; }
+
+    /// TODO(Subv): Find out what this is actually used for.
```
Comment (Subv) [inline src/core/hle/applets/erreula.h]: This comment is no longer relevant, see the other applet header files.

**Comment by mailwl** [comment]
done.

**Comment by Subv** [comment]
LGTM please squash your changes

**Comment by mailwl** [comment]
done, something wrong with travis :(

**Comment by Subv** [comment]
Can we get this merged already please?

**Comment by neobrain** [comment]
Hi, this is neobot, using neobrain's account. I'm keeping an archive of versions of this PR:

2016-07-04T06:55:23Z: https://github.com/citra-emu/citra/compare/b48b8ed5e5ac5fc025095b4dc220e9529eecb643...mailwl:e06f2705f07b71e34ce503b045f6a81e82db91e9

**Pull request merged**